### DISCUSSION TITLE
Rename quarkus-minikube to eg. quarkus-local-k8s

### DISCUSSION THREAD

**Discussion by llowinge** [quarkusio/quarkus#35027] [Community]
Following https://quarkus.io/guides/deploying-to-kubernetes we can notice that generated kubernetes files contain
```
"imagePullPolicy" : "Always"
```

which is probably OK on non-local kubernetes. But if we want to use local kubernetes (**Kubernetes from Docker for desktop on MacOS**) for playing/developing we must use workaround eg.
```
quarkus.kubernetes.image-pull-policy=IfNotPresent
```
otherwise it tries to pull the image from dockerhub even it is already present locally.

**This is really frustrating starting point for MacOS developers on local kubernetes.**

When i use `quarkus-minikube` it works. But i don't use minikube. Because of that i'm suggesting to rename the `quarkus-minikube` to some more generic name eg. `quarkus-local-k8s`.

**Comment by gsmet** [comment]
I'm not completely sure why you call `quarkus.kubernetes.image-pull-policy=IfNotPresent` a workaround? That's exactly what it is supposed to be doing.

What exactly is frustrating in having to add this property if you don't want to pull the images?

**Comment by llowinge** [comment]
@gsmet Users which need to add another configuration option (`quarkus.kubernetes.image-pull-policy=IfNotPresent`) to enable Quarkus on `local Kubernetes from Docker desktop` for simple use case can be seen as workaround (but maybe it is not). I don't know how to call it, but it doesn't definitely help to onboard new users with MacOS. I guess because of such "workarounds" the `quarkus-minikube` extension was created. Now it can solve the problem also with the `local Kubernetes from Docker desktop`, but the naming doesn't work anymore.

Frustrating thing is, that one has to find out what is the real problem first. As user will see failing pods. Then finding how to fix it.

**Comment by iocanel** [comment]
I'd say that `quarkus-minkube` was created as a developer experience enhancement for users that are new to the Kubernetes world and will most likely learn the hard way what image pull policy really is.

Personally, I don't mind a `quarkus-kubernetes-local'  as a base for all local kubernetes environments like:

- minikube
- microk8s
- kind
or whatever.

**Comment by llowinge** [comment]
@iocanel How can we move it forward ? I don't see into the process, should i create issue somewhere or this one is sufficient for tracking it ?
